File: nhom4_QLBA_DLH/lib/models/AppointmentDetails.dart
import 'package:nhom4_qlba_dlh/models/Doctor.dart';
import 'package:nhom4_qlba_dlh/models/Service.dart';

import 'package:nhom4_qlba_dlh/models/Doctor.dart';
import 'package:nhom4_qlba_dlh/models/Service.dart';

class AppointmentDetails {
  final int id;
  final String userName;
  final int doctorId;
  final int serviceId;
  final int appointmentId;
  final DateTime? appointmentDate;
  final DateTime createdAt;
  final String? appointmentDateStart;  // Thời gian bắt đầu (String)
  final String? appointmentDateEnd;    // Thời gian kết thúc (String)
  final Doctor? doctor;
  final Service? service;

  AppointmentDetails({
    required this.id,
    required this.userName,
    required this.doctorId,
    required this.serviceId,
    required this.appointmentId,
    this.appointmentDate,
    required this.createdAt,
    this.appointmentDateStart,
    this.appointmentDateEnd,
    this.doctor,
    this.service,
  });

  factory AppointmentDetails.fromJson(Map<String, dynamic> json) {
    return AppointmentDetails(
      id: json['id'],
      userName: json['userName'],
      doctorId: json['doctorId'],
      serviceId: json['serviceId'],
      appointmentId: json['appointmentId'],
      appointmentDate: json['appointmentDate'] != null
          ? DateTime.parse(json['appointmentDate'])
          : null,
      createdAt: DateTime.parse(json['createdAt']),
      appointmentDateStart: json['appointmentDateStart'],
      appointmentDateEnd: json['appointmentDateEnd'],
      doctor: json['doctor'] != null ? Doctor.fromJson(json['doctor']) : null,
      service: json['service'] != null ? Service.fromJson(json['service']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userName': userName,
      'doctorId': doctorId,
      'serviceId': serviceId,
      'appointmentId': appointmentId,
      'appointmentDate': appointmentDate?.toIso8601String(),
      'createdAt': createdAt.toIso8601String(),
      'appointmentDateStart': appointmentDateStart,
      'appointmentDateEnd': appointmentDateEnd,
      'doctor': doctor?.toJson(),
      'service': service?.toJson(),
    };
  }
}

File: Nhom4_QLBA_API/Nhom4_QLBA_API/Models/AppointmentDetails.cs
/*namespace Nhom4_QLBA_API.Models
{
    public class AppointmentDetails
    {
        public int Id { get; set; }
        public string? UserName { get; set; }
        //public int PatientId { get; set; }
        public int DoctorId { get; set; }
        public int ServiceId { get; set; }
        public int AppointmentId { get; set; }

        public DateTime? AppointmentDate { get; set; }
        public DateTime CreatedAt { get; set; }

        // Mối quan hệ với các bảng khác
        //public Patients? Patient { get; set; }
        public Doctors? Doctor { get; set; }
        public Services? Service { get; set; }
        public Appointments? Appointment { get; set; }
       
    }
}
*/

namespace Nhom4_QLBA_API.Models
{
    public class AppointmentDetails
    {
        public int Id { get; set; }
        public string? UserName { get; set; }
        public int DoctorId { get; set; }
        public int ServiceId { get; set; }
        public int AppointmentId { get; set; }

        public DateTime? AppointmentDate { get; set; }
        public DateTime CreatedAt { get; set; }

        // Thêm các thuộc tính mới để lưu thời gian bắt đầu và kết thúc
        public TimeSpan? AppointmentDateStart { get; set; }
        public TimeSpan? AppointmentDateEnd { get; set; }

        // Mối quan hệ với các bảng khác
        public Doctors? Doctor { get; set; }
        public Services? Service { get; set; }
        public Appointments? Appointment { get; set; }
    }
}
